src/event.cc

```diff
@@ -235,14 +228,12 @@ EventScheduler::checkEvents(int)
         ev_entry *event = tasks;
         assert(event);
 
+        const bool heavy = event->weight && !event->call->canceled();
+
         /* XXX assumes event->name is static memory! */
-        AsyncCall::Pointer call = asyncCall(41,5, event->name,
-                                            EventDialer(event->func, event->arg, event->cbdata));
-        ScheduleCallHere(call);
+        ScheduleCallHere(event->call);
 
         last_event_ran = event->name; // XXX: move this to AsyncCallQueue
-        const bool heavy = event->weight &&
-                           (!event->cbdata || cbdataReferenceValid(event->arg));
```

Comment:
This PR should probably continue to set `heavy` to false for heavy events with invalidated cbdata. Otherwise, we would need to debate why it is a good idea to degrade that existing support, which can be a painful debate because heavy events should not exist at all (and the PR does not add any). You should be able to reuse the same (new) code that you will need for EventScheduler::dump().

Reply:
That will depend on the parent CallDialer type providing a virtual canDial() method. Can you remind me why that method was not implemented?

If the reason does not apply still I will add it in a separate PR. It would resolve quite a few TODOs.

Reply:
> That will depend on the parent CallDialer type providing a virtual canDial() method.

I do not think so. Try implementing it without that method and, if you get stuck, I will help. Keep in mind that, in this context, only EventDialer calls can have invalid event cbdata (and you control EventDialer). For all other calls, just using AsyncCall::canceled() (which this PR code is already using) is enough to skip irrelevant heavy events.

Ignoring irrelevant heavy events may not be a big deal, but we need exactly the same code for reporting stale events in the queue anyway, and it is not difficult to add, so we should not remove that existing functionality in this PR.

> Can you remind me why that method was not implemented?

I think the best short answer is "because it is usually not needed and is difficult to add correctly". The current AsyncCall/Dialer split of responsibilities is wrong, but fixing that design bug requires lots of difficult code changes. I have tried fixing it, but had to postpone the project because there are more important things to do first. Moreover, C++20 might offer some overall better solutions instead. Just moving from type-specific CallDialer to a class with virtual functions will eat CPU cycles but will not solve the problems we need to solve (and is likely to make solving them more difficult). I can find my more specific notes if you insist, but it will take time, and the subject is not really relevant to _this_ PR.

> If the reason does not apply still I will add it in a separate PR. It would resolve quite a few TODOs.

Please do not add it. AFAICT, the underlying problem is a much more difficult problem to fix than you realize.

Reply:
Okay, I will leave as-is.

The problem with your initial request then -  is that the cbdata arg reference the old manager report showed (aka. Job alive/dead status) is now hidden inside the Dialer. Without a canDial() to fetch the reported info regardless of Dialer type we have to change the column meaning.

I have considered making this column show the canceled() status of the Call. But in light of your request to drop canceled() Calls from the queue it would be a column of only "yes" entries with no meaning.

Reply:
Done in 690ba6d